nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/learning/config/AdaBelief.java

```diff
@@ -0,0 +1,132 @@
+/*
+ *  ******************************************************************************
+ *  *
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Apache License, Version 2.0 which is available at
+ *  * https://www.apache.org/licenses/LICENSE-2.0.
+ *  *
+ *  *  See the NOTICE file distributed with this work for additional
+ *  *  information regarding copyright ownership.
+ *  * Unless required by applicable law or agreed to in writing, software
+ *  * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *  * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *  * License for the specific language governing permissions and limitations
+ *  * under the License.
+ *  *
+ *  * SPDX-License-Identifier: Apache-2.0
+ *  *****************************************************************************
+ */
+
+package org.nd4j.linalg.learning.config;
+
+import lombok.Builder;
+import lombok.Data;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.nd4j.linalg.learning.AdaBeliefUpdater;
+import org.nd4j.linalg.learning.GradientUpdater;
+import org.nd4j.linalg.schedule.ISchedule;
+import org.nd4j.shade.jackson.annotation.JsonProperty;
+
+import java.util.Arrays;
+import java.util.Map;
+
+/**
+ * AdaBelief 
+ * https://arxiv.org/pdf/2010.07468.pdf
+ */
+@Data
+@Builder(builderClassName = "Builder")
+public class AdaBelief implements IUpdater {
+
+    public static final double DEFAULT_LEARNING_RATE = 1e-3;
+    public static final double DEFAULT_EPSILON = 1e-8;
```

Comment:
The paper authors have found that a smaller default epsilon works better:

> Epsilon in AdaBelief is different from Adam (typically eps_adabelief = eps_adam*eps_adam)
> ( eps of Adam in Tensorflow is 1e-7, in PyTorch is 1e-8, need to consider this when use AdaBelief in Tensorflow)

So I guess we might want to use either 1e-14 or 1e-16 here.

Reply:
changed to 1e-14